### PR TITLE
support for creating only a reset partition

### PR DESCRIPTION
**Pull request by mwhudson**
This is not very pretty but it works, and perhaps isn't toooooo bad?

### CODE REVIEW

**Review by dbungert — COMMENTED** [review]
I think this is fine but I'd like some sort of test automation on this.  Perhaps like what's going on in subiquity/server/controllers/tests/test_filesystem.py TestGuidedV2?  We also have some autoinstall tests going on in the API tests, either of those could probably do the trick.

**Comment by mwhudson** [comment]
Bah the new test is failing. It works on my machine!

File: scripts/validate-yaml.py [hunk 1]
```diff
@@ -75,11 +75,15 @@ def final_checks(self):
...
 
-
-
 def main():
     storage_checker = StorageChecker()
 
+    final_checks = True
+
+    if sys.argv[1:2] == ['--no-root-mount']:
+        final_checks = False
```
Comment (dbungert) [inline scripts/validate-yaml.py:84]: nitpick: the connection to the generic `final_checks` and this arg name feels non-obvious.  Suggest renaming to `root_mount`.

File: subiquity/server/controllers/install.py [hunk 2]
```diff
@@ -259,7 +259,14 @@ async def run_curtin_step(name, stages, step_config, source=None):
 
         await run_curtin_step(name="initial", stages=[], step_config={})
 
-        if fs_controller.is_core_boot_classic():
+        if fs_controller.reset_partition_only:
```
Comment (dbungert) [inline subiquity/server/controllers/install.py:262]: This method is getting a bit long.  I suggest we start splitting this method up, maybe by creating an install type and doing single dispatch stuff.  Doesn't have to be this PR.

**Review by dbungert — APPROVED** [review]

File: subiquity/server/controllers/install.py (hunk 2 above)
Comment (mwhudson) [reply subiquity/server/controllers/install.py:262]: yeah, it's a long series of involved function calls that are very similar but not quite the same

**Pull request merged**